### PR TITLE
Adding switch to UpdatePackageIndex Task that will Update stable versions

### PR DESCRIPTION
**Pull request by joperezr**
cc: @ericstj 

Contributes to https://github.com/dotnet/corefx/issues/39537

Add switch to UpdatePackageIndex that will fetch stable package versions from nuget feeds and update the package index with that information.

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/UpdatePackageIndex.cs [hunk 1]
```diff
@@ -73,8 +76,20 @@ public class UpdatePackageIndex : BuildTask
         /// </summary>
         public ITaskItem[] InboxFrameworkLayoutFolders { get; set; }
 
+        /// <summary>
+        /// Version endpoints to be queried in order to get the latest stable patch
```
[the remark this answers is not in the document]
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/src/UpdatePackageIndex.cs]: fixed it, now it's using the settings that dotnet restore would use.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/UpdatePackageIndex.cs [hunk 2]
```diff
@@ -11,12 +12,14 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using static Microsoft.DotNet.Build.Tasks.Packaging.ValidateHarvestVersionIsLatestForEra;
```
[the remark this answers is not in the document]
Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/src/UpdatePackageIndex.cs]: fixed.

**Comment by joperezr** [comment]
I'll go ahead and merge this in since we have to get it for 3.0 so we need to consume this in master first. I will address any additional feedback that comes out from this in a subsequent PR.

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetUtility.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            var provider = new PackageSourceProvider(settings);
+            return provider.LoadPackageSources().Where(e => e.IsEnabled == true).ToList();
+        }
+
+        public static Version GetLatestPatchStableVersionForRelease(this IEnumerable<Version> versions, int eraMajorVersion, int eraMinorVersion)
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetUtility.cs]: More instances of era here

File: src/Microsoft.DotNet.Build.Tasks.Packaging/src/UpdatePackageIndex.cs [hunk 4]
```diff
@@ -180,6 +185,17 @@ public override bool Execute()
                 }
             }
 
+            if (UpdateStablePackageInfo && Packages == null && PackageFolders == null)
```
Comment (ericstj) [inline src/Microsoft.DotNet.Build.Tasks.Packaging/src/UpdatePackageIndex.cs]: Do you get null or empty array here?  Make sure you can actually trigger this codepath by just specifying UpdateStablePackageInfo=true without any other params.

Comment (joperezr) [reply src/Microsoft.DotNet.Build.Tasks.Packaging/src/UpdatePackageIndex.cs]: I get null, I did test this out and ensured it work by just setting the one property.

**Pull request merged**